src/main/java/net/server/channel/handlers/WhisperHandler.java

```diff
@@ -26,121 +26,85 @@
+    public static final byte RT_DIFFERENT_CHANNEL = 0x03;
+
     @Override
-    public final void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c) {
-        byte mode = slea.readByte();
-        if (mode == 6) { // whisper
-            String recipient = slea.readMapleAsciiString();
-            String text = slea.readMapleAsciiString();
-            MapleCharacter player = c.getChannelServer().getPlayerStorage().getCharacterByName(recipient);
-            if (c.getPlayer().getAutobanManager().getLastSpam(7) + 200 > currentServerTime()) {
-                return;
-            }
-            if (text.length() > Byte.MAX_VALUE && !player.isGM()) {
-            	AutobanFactory.PACKET_EDIT.alert(c.getPlayer(), c.getPlayer().getName() + " tried to packet edit with whispers.");
-            	FilePrinter.printError(FilePrinter.EXPLOITS + c.getPlayer().getName() + ".txt", c.getPlayer().getName() + " tried to send text with length of " + text.length());
-            	c.disconnect(true, false);
-            	return;
-            }
-            if (player != null) {
-                player.getClient().announce(MaplePacketCreator.getWhisper(c.getPlayer().getName(), c.getChannel(), text));
-                if (YamlConfig.config.server.USE_ENABLE_CHAT_LOG) {
-                    LogHelper.logChat(c, "Whisper To " + player.getName(), text);
-                }
-                if(player.isHidden() && player.gmLevel() >= c.getPlayer().gmLevel()) {
-                    c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 0));
-                } else {
-                    c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 1));
-                }
-            } else {// not found
-                World world = c.getWorldServer();
-                    if (world.isConnected(recipient)) {
-                        world.whisper(c.getPlayer().getName(), recipient, c.getChannel(), text);
-                        if (YamlConfig.config.server.USE_ENABLE_CHAT_LOG) {
-                            LogHelper.logChat(c, "Whisper To " + recipient, text);
-                        }
-                        player = world.getPlayerStorage().getCharacterByName(recipient);
-                        if(player.isHidden() && player.gmLevel() >= c.getPlayer().gmLevel())
-                            c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 0));
-                        else
-                            c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 1));
-                    } else {
-                        c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 0));
-                    }
+    public void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c) {
+        byte request = slea.readByte();
+        String name = slea.readMapleAsciiString();
+        MapleCharacter target = c.getWorldServer().getPlayerStorage().getCharacterByName(name);
+
+        if (target != null) {
```

Comment:
It's cleaner to just do a null check, give error, and return early. It's used in other places for anti cheat checks so it should be fine for standards.